active/0021-auto-subscribe.md

```diff
@@ -0,0 +1,86 @@
+# EMQX Auto Subscribe
+
+## Changelog
+
+* 2022-07-06: @DDD
+
+## Abstract
+
+With Auto Subscription enabled, after a client is successfully connected to EMQX, it may not need to send SUBSCRIBE requests to EMQX, because EMQX will complete the pre-defined subscriptions for the client.
+
+Prior to version 5, this feature is also known as "proxy subscription".
+
+## Motivation
+
+1. Missing subscription option (rh rap nl) for MQTT v5 version in table structure.
+2. Lack of support for HTTP services.
+3. Relies on the rules engine, which essentially acts as a hook.
+4. Does not support placeholders, does not support variables, that is, the database is stored topics only.
+5. The table structure is fixed and immutable. Only `clientid` is supported as primary key.
+
+## Require
+
+1. Support MQTT v5 sub options.
+2. Get subscription from HTTP service.
```

Comment:
do we really want to support this with an HTTP backend ?

Reply:
I would prefer a set of mgmt api for subscription instead of supporting 3rd parity http backend.